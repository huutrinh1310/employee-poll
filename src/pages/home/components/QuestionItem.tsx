import { Button } from "@/components/ui/button";
import {
  Card,
  CardContent,
  CardDescription,
  CardHeader,
  CardTitle,
} from "@/components/ui/card";
import { Question } from "@/types/entities.type";
import dayjs from "dayjs";
import { useNavigate } from "react-router-dom";

export interface IQuestionItemProps {
  data?: Question;
}

export default function QuestionItem({ data }: IQuestionItemProps) {
  const navigate = useNavigate();
  return (
    <Card className="w-full md:w-[400px] lg:w-[300px]">
      <CardHeader>
        <CardTitle>{data?.author}</CardTitle>
        <CardDescription>
          {dayjs(data?.timestamp).format("hh:mm A | DD/MM/YYYY")}
        </CardDescription>
      </CardHeader>
      <hr />
      <CardContent className="pt-3">
        <Button
          className="w-full border-primary"
          variant={"outline"}
          children={"Show"}
          onClick={() => {
            navigate("/questions/" + data?.id);
          }}
        />
      </CardContent>
    </Card>
  );
}
