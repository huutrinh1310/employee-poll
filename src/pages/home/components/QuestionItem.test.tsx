import { expect, test, describe } from "@jest/globals";
import { render } from "@testing-library/react";
import QuestionItem from "@/pages/home/components/QuestionItem";
import { MemoryRouter } from "react-router-dom";

describe("QuestionItem", () => {
  test("matches the snapshots when a question is passed", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <QuestionItem
          data={{
            id: "abc",
            author: "sarahedo",
            timestamp: Date.now(),
            optionOne: {
              votes: ["sarahedo"],
              text: "Build our new application with Javascript",
            },
            optionTwo: {
              votes: [],
              text: "Build our new application with Typescript",
            },
          }}
        />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  test("matches the snapshots when a question is passed", () => {
    const { asFragment } = render(
      <MemoryRouter>
        <QuestionItem />
      </MemoryRouter>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
