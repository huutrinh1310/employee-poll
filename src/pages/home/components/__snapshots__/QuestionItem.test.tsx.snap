// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuestionItem matches the snapshots when a question is passed 1`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full md:w-[400px] lg:w-[300px]"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="text-2xl font-semibold leading-none tracking-tight"
      >
        sarahedo
      </h3>
      <p
        class="text-sm text-muted-foreground"
      >
        11:32 AM | 20/08/2024
      </p>
    </div>
    <hr />
    <div
      class="p-6 pt-3"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full border-primary"
      >
        Show
      </button>
    </div>
  </div>
</DocumentFragment>
`;

exports[`QuestionItem matches the snapshots when a question is passed 2`] = `
<DocumentFragment>
  <div
    class="rounded-lg border bg-card text-card-foreground shadow-sm w-full md:w-[400px] lg:w-[300px]"
  >
    <div
      class="flex flex-col space-y-1.5 p-6"
    >
      <h3
        class="text-2xl font-semibold leading-none tracking-tight"
      />
      <p
        class="text-sm text-muted-foreground"
      >
        11:32 AM | 20/08/2024
      </p>
    </div>
    <hr />
    <div
      class="p-6 pt-3"
    >
      <button
        class="inline-flex items-center justify-center whitespace-nowrap rounded-md text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border bg-background hover:bg-accent hover:text-accent-foreground h-10 px-4 py-2 w-full border-primary"
      >
        Show
      </button>
    </div>
  </div>
</DocumentFragment>
`;
